Framework/DataHandling/src/LoadBBY.cpp

```diff
@@ -498,21 +515,40 @@ void LoadBBY::loadInstrumentParameters(
         // comma separated details
         std::vector<std::string> details;
         boost::split(details, line, boost::is_any_of(","));
+        if (details.size() < 3) {
+          g_log.warning() << "Invalid format for BILBY parameter " << x.first
+                          << std::endl;
+          continue;
+        }
         auto hdfTag = boost::algorithm::trim_copy(details[0]);
         try {
-          // extract the parameter and add it to the parameter dictionary
-          if (!hdfTag.empty() && loadNXDataSet(entry, hdfTag, tmpFloat)) {
-            auto factor = std::stod(details[1]);
-            logParams[logTag] = factor * tmpFloat;
-          } else if (details.size() > 2) {
-            logParams[logTag] = std::stod(details[2]);
+          // extract the parameter and add it to the parameter dictionary,
+          // check the default for value numeric and string
+          auto updateOk = false;
+          if (!hdfTag.empty()) {
+            if (isNumeric(details[2])) {
+              if (loadNXDataSet(entry, hdfTag, tmpFloat)) {
+                auto factor = std::stod(details[1]);
+                logParams[logTag] = factor * tmpFloat;
+                updateOk = true;
+              }
+            } else {
+              if (loadNXString(entry, hdfTag, tmpString)) {
+                logStrings[logTag] = tmpString;
+                updateOk = true;
+              }
+            }
+          }
+          if (!updateOk) {
+            if (isNumeric(details[2]))
+              logParams[logTag] = std::stod(details[2]);
+            else
+              logStrings[logTag] = details[2];
             if (!hdfTag.empty())
-              g_log.warning() << "Cannot find hdf parameter "
-
-                              << hdfTag << ", using default.\n";
+              g_log.warning() << "Cannot find hdf parameter " << hdfTag
+                              << ", using default.\n";
           }
-
-        } catch (const std::invalid_argument &) {
+        } catch (std::invalid_argument &) {
```

Comment:
Why was the `const` removed? There seems to be a mixture of styles in this file but it's usually better to pass by const reference unless there's a reason not to.

Reply:
My slip up - restored the const.